File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/model/flat/Province.java
package org.babyfish.jimmer.sql.model.flat;

import org.babyfish.jimmer.sql.*;
import org.jspecify.annotations.Nullable;

import java.util.List;

@Entity
public interface Province {

    @Id
    long id();

    String provinceName();

    @Nullable
    @ManyToOne
    @OnDissociate(DissociateAction.DELETE)
    Country country();

    @OneToMany(mappedBy = "province")
    List<City> cities();
}

File: project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/model/flat/Province.kt
package org.babyfish.jimmer.sql.kt.model.flat

import org.babyfish.jimmer.sql.DatabaseValidationIgnore
import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.ManyToOne
import org.babyfish.jimmer.sql.Table

@Entity
@Table(name = "FLAT_PROVINCE")
@DatabaseValidationIgnore
interface Province {

    @Id
    val id: Long

    val provinceName: String

    @ManyToOne
    val country: Country
}